src/js/player.js

```diff
@@ -1747,10 +1747,11 @@ class Player extends Component {
    */
   duration(seconds) {
     if (seconds === undefined) {
-      return this.cache_.duration || 0;
+      // return NaN if the duration is not known
+      return this.cache_.duration !== undefined ? this.cache_.duration : NaN;
     }
 
-    seconds = parseFloat(seconds) || 0;
+    seconds = parseFloat(seconds);
```

Comment:
I think in here we still want to do `|| 0` because if what is passed in, isn't a number we should just assume 0.